plugins/out_s3/s3.c

```diff
@@ -1035,6 +1056,7 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
     }
 
     ret = s3_put_object(ctx, tag, create_time, body, body_size);
+    flb_free(payload_buf);
```

Comment:
So if we're in the case where the data is not compressed, we will call free on a NULL pointer, but that's ok.

Reply:
got it, will wrap it with compress only